src/util/upload-plan-parser.ts

```diff
@@ -21,32 +22,47 @@ class UploadPlanParser {
         }
     }
 
-    static convertToDownloadFilesJob(job: Job, downloadDirBasePath: string): DownloadFilesJob {
+    static convertToDownloadFilesJob(job: Job, downloadDirBasePath: string): DownloadS3FilesJob {
         const filesToDownload: File[] = job.conversion ? job.conversion.inputs : job.files;
 
-        const downloadJob: DownloadFilesJob = {
+        return {
             basePath: downloadDirBasePath,
             container: job.manifest_id,
             files: UploadPlanParser.parseDownloadFiles(filesToDownload)
-        };
+        }
+    }
 
-        return downloadJob
+    static convertToDownloadBundleFilesJob(job: Job, basePath: string): DownloadBundleFilesJob {
+        const filesToDownload: File[] = job.conversion ? job.conversion.inputs : job.files;
+        if (job.dcp_bundle_uuid) {
+            return {
+                basePath: basePath,
+                bundleUuid: job.dcp_bundle_uuid,
+                container: job.manifest_id,
+                files: R.map((file :File) => { return { fileName: file.name } }, filesToDownload)
+            };
+        } else {
+            throw new TypeError("Cannot create DCP Download Job without DCP Bundle UUID")
+        }
     }
 
     static convertToConvertFilesJob(job: Job, fileDirBasePath: string): ConvertFilesJob {
-        const convertFilesJob: ConvertFilesJob = {
+        return {
             reads: UploadPlanParser.parseConvertFiles(job.conversion!.inputs),
             outputName: job.conversion!.output_name,
             outputDir: `${fileDirBasePath}/${job.manifest_id}`
-        }
-        return convertFilesJob;
+        };
     }
 
     static parseDownloadFiles(files: File[]): DownloadFile[] {
         return R.map((file) => {
+            let source = "";
```

Comment:
Is this change needed? Can't see any other changes related to the source field here. If so, how do we handle an empty source?

Reply:
The source will/can be empty in the case of DCP files.